libraries/node-core-library/src/FileSystem.ts

```diff
@@ -785,6 +791,70 @@ export class FileSystem {
     });
   }
 
+  /**
+   * Writes the contents of multiple Uint8Arrays to a file on disk, overwriting the file if it already exists.
+   * Behind the scenes it uses `fs.writevSync()`.
+   *
+   * This API is useful for writing large files efficiently, especially if the input is being concatenated from
+   * multiple sources.
+   *
+   * @remarks
+   * Throws an error if the folder doesn't exist, unless ensureFolder=true.
+   * @param filePath - The absolute or relative path of the file.
+   * @param contents - The content that should be written to the file.
+   * @param options - Optional settings that can change the behavior.
+   */
+  public static writeBuffersToFile(
+    filePath: string,
+    contents: ReadonlyArray<Uint8Array>,
+    options?: IFileSystemWriteBinaryFileOptions
+  ): void {
+    FileSystem._wrapException(() => {
+      // Need a mutable copy of the iterable to handle incomplete writes,
+      // since writev() doesn't take an argument for where to start writing.
+      const toCopy: Uint8Array[] = [...contents];
+
+      let fd: number | undefined;
+      try {
+        fd = fsx.openSync(filePath, 'w');
+      } catch (error) {
+        if (!options?.ensureFolderExists || !FileSystem.isNotExistError(error as Error)) {
+          throw error;
+        }
+
+        const folderPath: string = nodeJsPath.dirname(filePath);
+        FileSystem.ensureFolder(folderPath);
+        fd = fsx.openSync(filePath, 'w');
+      }
+
+      let position: number = 0;
+      let buffersToSkip: number = 0;
```

Comment:
It would be more clear to allocate `buffersToSkip` on line 841, as it is always guaranteed to be zero there.  And then you don't need to reset it to zero on 849.